python-api/src/main/python/api/client.py

```diff
@@ -0,0 +1,47 @@
+#
+# Licensed to Cloudera, Inc. under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  Cloudera, Inc. licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from abc import ABCMeta, abstractmethod
+from urlparse import urlparse
+
+class Client:
+    __metaclass__ = ABCMeta
+
+    @abstractmethod
+    def submit(self, context):
```

Comment:
This comments also applies to your Scala change, but you really should write documentation for public methods (scaladoc / javadoc in Scala, doc strings in Python). Especially here, where you're creating a new API (even if based on an existing one), that can help you figure out issues in the API.

For example, what is "context"? Down below, why is there a default value for `uri` and why is it the `urlparse` module? Why is there a default value for `input_file` and why is it a `file`?